app/cmd/rpc/server.go

```diff
@@ -60,15 +60,43 @@ func StartRPC(port string, timeout int64, simulation, debug, allBlockTxs, hotRel
 		ReadHeaderTimeout: 20 * time.Second,
 		WriteTimeout:      60 * time.Second,
 		Addr:              ":" + port,
-		Handler:           http.TimeoutHandler(Router(routes), time.Duration(timeout)*time.Millisecond, "Server Timeout Handling Request"),
+		Handler:           http.TimeoutHandler(Router(routes, rpcMaxBytes), time.Duration(timeout)*time.Millisecond, "Server Timeout Handling Request"),
 	}
 	log.Fatal(srv.ListenAndServe())
 }
 
-func Router(routes Routes) *httprouter.Router {
+func enforceMaxBytesPayload(handler httprouter.Handle, maxBodySize int64) httprouter.Handle {
+	if maxBodySize == 0 {
+		maxBodySize = types.DefaultRPCMaxBytesSize
+	}
+
+	return reuseBody(func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
```

Comment:
Not too familiar with reuseBody. If we implement max payload, does that mean everytime for every request now, we are copying the body to read it again?

Seems like an performance overhead unless im a mistaken..

Reply:
the short answer is yes, the only way to read body 2 times is duplicate it, to avoid that i should change/apply the change over a lot of places in core code where they unmarshal it.
it was mainly to remove the hardcoded value and allow it to be configurable, I think anyway it should not even be handled here and should be handled on the proxy that it need to have in front.
anyway I do not think it will drop the performance in a way that affect the service.